client/http_sender.go

```diff
@@ -125,14 +125,11 @@ func (s *httpSender) Send(_ context.Context, call Call) {
 				// the errors we'll sweep up in this net shouldn't be retried,
 				// but we can't really know for sure which.
 				log.Warningf("failed to send HTTP request or read its response: %s", t)
-				return retry.Continue, nil
-			default:
-				// Can't retry in order to recover from this error. Propagate.
-				return retry.Break, err
+				return retry.Continue, err
 			}
 		}
-		// On successful post, we're done with retry loop.
-		return retry.Break, nil
+		// If not previously handled, propagate the error (which may be nil).
```

Comment:
On the other hand, since this has been a problem in practice (while `return nil` in a non-retryable function has not), there does seem to be an issue with the dual use of `retry.Break`. It sounds like an error so people don't think they need another return value. Maybe we should add a new value for `retry.Status`: `retry.Success` would panic if returned with a non-nil error, and `retry.Break` would then panic with a nil error.